### DISCUSSION TITLE
Website cleanup

### DISCUSSION THREAD

**Discussion by tbreuss** [MithrilJS/mithril.js#2834] [General]
# Website cleanup

Since there have been discussions about refreshing the website design for a long time, but it does not really move forward, I have been working on a website cleanup branch. Mainly, because some things are a bit embarrassing.

The idea behind this cleanup is:

- keep the current web design, but make it more appealing
- fix all the obvious design/UI/UX issues
- fix obviously wrong content
- add or refresh content where necessary

I see this as a foundation for the next bigger web design update.

- Preview: https://www.mithriljs.tebe.ch
- Branch: https://github.com/tbreuss/mithril.js/tree/feat/doc-enhancements
- Pull Request: https://github.com/MithrilJS/mithril.js/pull/2831

Please note that this is only a minor update to the current website design. If you want to see the differences, you might have to open both versions with your browser and compare them visually.

What you all think about it?

---

**The branch includes the following changes so far**:

- [x] link logo with homepage (better ux imho, and it is simply expected)
- [x] set overall line-height to 1.5 (more appealing)
- [x] fix all broken anchor links from menu and within content (this was annoying)
- [x] open all external links in a new tab (better ux imho)
- [x] in menu remove intendation of first level, add margin bottom to section (more space for 3rd level menu items)
- [x] reduce font size for code examples highlighted by prisma.js to same size as flems examples (so, font-sizes of normal preformated text, code examples and flems examples are the same size)
- [x] increase margin bottom for code examples to approx. 150% of p-tags (more appealing)
- [x] remove all "how to contribute" menu items und place it under the page as table of contents (those menu item link labels were way too long)
- [x] remove "code" formattings in table cells except for the description (makes it more readable)
- [x] for all backgrounds use a similar color tone, the one from prisma.js (so we have a simple color scheme)
- [x] made the hamburger a bit larger (better ux)
- [x] fixed z-index issue with flems example laying on top of the menu layer (obvious ux issue) 
- [x] fixed issue with different line heights in paragraphs and other tags when using "code" tags (keep consistent line-height)
- [x] in prisma.js fixed wrong background color for operators (looks to me like a bug in prisma.js)
- [x] use "white-space:nowrap" and "text-overflow:ellipsis" for left menu items (more appealing, better readability)
- [x] remove weird looking link labels when using "code" within "a" tags (this was terrible, much better now with better readability)
- [x] remove weird looking link labels when using "code" within "a" within "h" tags (same as above)
- [x] add mouse-over for "a>code"-tag (makes it clearer that it is a link)
- [x] adjust indentation of ordered and unordered lists (better design and imho cleaner)
- [x] break line of very long \<code\> examples within p-tags on small screen (broke the layout)
- [x] remove link to "jsonp" page (page was removed at some point)
- [x] on small screen hide first ruler after h1 consequently (some pages have a ruler, some not, depending wether there are menu items for the page, unnecessary inconsistency)
- [x] cleanup header hierarchy on certain pages (code-of-conduct, contributing, integrating-libs, releasing)
- [x] add a propper 404 page (nicer for the visior, with additional help maybe)
- [x] remove all release candidate versions from the version selector (not needed, as each version can be accessed by entering the URL in the address bar)
- [x] remove Mithril.js Jobs from Misc section (linked wiki page is really outdated)
- [x] reordered the "Misc" section (more "logical" now)
- [x] move `ES6+ on legacy browsers` from Resources to Misc section (to make it less prominent)
- [x] move `Learning Resources` from Getting Started to Misc section (to make it less prominent)
- [x] remove `v1 Documentation` and `v0.2 Documentation` from Misc section (we have those links already in the version selector besides the logo, and we don't want to promote older versions, of course)
- [x] cleanup left menu items
- [x] Change footer "License: MIT. © Leo Horie." to something more appropriate (better and current information with link to community)
- [x] add sticky footer (looks better on pages with little content)
- [x] add smooth scroll behavior (it's nicer) 
- [x] #2833
- [x] #2832 
- [x] remove "[ES6+ on legacy browsers](https://www.mithriljs.tebe.ch/es6.html#es6+-on-legacy-browsers) from menu (@modir). Instead removing we moved it to the misc section
- [x] slightly lighter base font color #181818 (looks lighter)
- [x] memorize scroll position of left column across page loads (better ux) 
- [x] bump prism.js up to latest 1.29.0 version
- [x] make two col layout with 100% height and scrollbars (in progress)
- [x] add "two columns 100% height layout with scrollbars" like here: https://mithriljs.tebe.ch/draft-01/
- [x] version selector: all we really need is latest v1.x and latest v2.x. Not sure it's worth listing minor versions (@dead-claudia)

---

**To do**:
- [ ] move all examples to docs
- [ ] inconsistent use of headers (i.e. contributing.html where we use h2 instead of h3)
- [ ] We are using header from h1 to h6 (on page stream.html). h6 is not even formatted. too much levels imho.
- [ ] question: remove \<hr\> ruler completely? we have well structured content, therefore it is not needed. Some pages have rulers, others not (paths.html for example)
- [ ] make \<hr\> ruler a bit nicer, with consistent vertical margin
- [ ] upgrade everything to modern JS (@dead-claudia)
- [ ] remove the text above Introduction, see comment in this discussion (@modir)
- [ ] on the [tutorial](https://www.mithriljs.tebe.ch/simple-application.html) add sub titles (@modir)
- [ ] integration chapters Components, Routing and XHR from the Introduction page into the tutorial (@modir)

---

**To discuss**:

(none so far)

---

### Heading Hierarchy

We are using:

- h1: 57
- h2: 48
- h3: 263
- h4: 250
- h5: 30
- h6: 2

**Comment by modir** [comment]
Thank you so far. I like all the changes so far.

Another question is: Do we still need the page "[ES6+ on legacy browsers](https://www.mithriljs.tebe.ch/es6.html#es6+-on-legacy-browsers)" or can we remove it as well from the menu on the left.

**Comment by modir** [comment]
And on the [tutorial](https://www.mithriljs.tebe.ch/simple-application.html) I would add sub titles.

Plus the chapters Components, Routing and XHR from the Introduction page I would migrate into the tutorial as well.

**Comment by modir** [comment]
And my last comment for now:
In the menu we have once "Getting Started" as a text above Introduction and then as a chapter within Introduction. I personally would remove the text above Introduction.

**Comment by tbreuss** [comment]
Thanks for your feedback, @modir. Lets wait for the seniors to hear their opinions. In the meantime I collect all the input...

**Comment by tbreuss** [comment]
Maybe we could use https://www.npmjs.com/package/marked-admonition-extension extension or use it manually like:

~~~js
var elms = document.evaluate( "//p[contains(., 'NOTE: ')]", document.body, null, XPathResult.UNORDERED_NODE_SNAPSHOT_TYPE, null)
for ( var i=0 ; i < elms.snapshotLength; i++ ){
    elms.snapshotItem(i).classList.add('note')
}
~~~

~~~css
.note {
    color: #856404;
    background-color: #fff3cd;
    border-color: #ffeeba;
    padding: 0.75em 1.25em;
}
.note code {
    background-color: inherit;
}
~~~

**Comment by cavemansspa** [comment]
nice work here @tbreuss.
